engine/flink/avro-util/src/main/scala/pl/touk/nussknacker/engine/avro/encode/ValidationMode.scala

```diff
@@ -0,0 +1,17 @@
+package pl.touk.nussknacker.engine.avro.encode
+
+case class ValidationMode private(name: String, label: String, acceptUnfilledOptional: Boolean, acceptRedundant: Boolean)
+
+object ValidationMode {
+
+  val strict: ValidationMode = ValidationMode("strict", "Strict mode", acceptUnfilledOptional = false, acceptRedundant = false)
+
+  val allowOptional: ValidationMode = ValidationMode("allowOptional", "No optional parameters validations", acceptUnfilledOptional = true, acceptRedundant = false)
+
+  val allowRedundantAndOptional: ValidationMode =  ValidationMode("allowRedundantAndOptional", "Allow redundant parameters", acceptUnfilledOptional = true, acceptRedundant = true)
+
+  val values: List[ValidationMode] = List(strict, allowOptional, allowRedundantAndOptional)
+
+  def byName(name: String): Option[ValidationMode] = values.find(_.name == name)
```

Comment:
I don't think that byName will be a good idea to identify mode, because we can change this label. WDYT about having in parameters value `(acceptUnfilledOptional,acceptRedundant)` or just  `acceptUnfilledOptional,acceptRedundant`?

Reply:
currently I have name and label to avoid such problems :)

Reply:
Indeed I see now. IMO `acceptUnfilledOptional,acceptRedundant` will be better because it will be easier to migrate such value in case when our validation approach will slightly changed e.g. one flag will be replaced by another or we will add other combinations of modes ... but it is also fine :+1: